message_processor.go

```diff
@@ -91,7 +86,7 @@ func sendMessage(l logger.Logger, buf *logdriver.LogEntry, t *partialMsgBuffer,
 
 // shouldSendMessage() returns a boolean indicating
 // if the message should be sent to Splunk
-func shouldSendMessage(message []byte) bool {
+func (mg messageProcessor) shouldSendMessage(message []byte) bool {
```

Comment:
I don't think it is an error though. For empty strings, it shouldn't be sent because HEC endpoint doesn't accept it. We just filter out the messages.

Reply:
@sharonx The error response can be propagated up to let the user (or a program) know to not send an empty/invalid message. This was what Ken suggested to me for the partial buffer append as well. I think it applies here. But it's not a stopper for me. You can merge if you still think its not needed. I'm good otherwise.

Reply:
I checked the partialLogBuffer::append(). I think it is a different case. In append(), there is an error (not able to append to the buffer due to program or system fault) happened and maybe the caller should handle it. Here we don't have any errors that's thrown from the program. It's a decision we make to fulfill HEC requirement.